js/calendar.js

```diff
@@ -83,3 +91,153 @@ function refreshCalendar(nextState) {
+			endDate = endDay + space + months[endMonth] + space + endYear;
+			if (!allDay) {
+				// "dd MMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startMonth != endMonth) {
+			// same year, different month - "dd MMM - dd MMM"
+			startDate = startDay + space + months[startMonth];
+			endDate = endDay + space + months[endMonth];
+			if (!allDay) {
+				// "dd MMM HH:MM - dd MMM HH:MM"
+				startDate += space + startHour + colon + startMinute;
+				endDate += space + startHour + colon + startMinute;
+			}
+		} else if (startDay != endDay) {
+			// same year, same month, different day - "dd - dd MMM YYYY"
+			if (allDay) {
+				startDate = startDay;
+				endDate = endDay + space + months[endMonth] + space + endYear;
+			} else {
+				// "dd MMMM YYYY HH:MM - dd MMM YYYY HH:MM"
+				startDate = startDay + space + months[startMonth] + space + startHour + colon + endMinute;
+				endDate = endDay + space + months[endMonth] + space + endHour + colon + endMinute;
+			}
+		} else {
+			// same day - "dd MMM HH:MM - HH:MM"
+			startDate = startDay + space + months[startMonth] + space + startHour + colon + startMinute;
+			endDate = endHour + colon + endMinute;
+
+		}
+		date = startDate + dash + endDate;
+
+		//	location
+		location = eventDataExtended.location;
+
+		// description 
+		description = parseDescription(eventDataExtended.description);
+
+		// Facebook event button
+		let eventFacebookLinkElement = document.getElementById("facebook-link");
+		let link = parseFacebookLink(eventDataExtended.description);
+		if (link) {
+			eventFacebookLinkElement.onclick = () => {window.open(link)};
+			eventFacebookLinkElement.style.display = "inline";
+		} else {
+			// catches null regex matches
+			eventFacebookLinkElement.style.display = "none";
+		}
+	
+
+		// set text
+		document.getElementById("event-text-title").textContent = title;
+		document.getElementById("event-text-date").textContent = date;
+		document.getElementById("event-text-location").textContent = location;
+		var descStr = "";
+		for (line in description) {
+			descStr += description[line].toString() + "<br>";
+		}
+		document.getElementById("event-text-description").innerHTML = descStr; //innerhtml since description is already string
+		document.getElementById('calendar-event').style.display = "block"; //show
+	} else {
+		//null event
+		console.log("ERROR: event is null");
+		//should I keep this?
+	}
+	
+}
+
+function hideEvent() {
+	// hide event information by hiding the element
+	document.getElementById('calendar-event').style.display = "none";
+}
+
+function parseDescription(description) {
+	var retDesc = "";
+	let regex = new RegExp(/&lt;desc&gt;(.*?)&lt;\/desc\&gt;/, "g");
+	// parse and escape for <desc></desc> tags. toString prevent XSS
+	let match = regex.exec(description.toString());
+	if (match != null) {
+		if (match.length >= 2) {
```

Comment:
what if match.length < 2? Is that a case that can happen? If it is, we should handle it, it not, we can remove this check.

Reply:
The grouping seems to make it that if it matches then the group (bit inside the tags) are the 2nd item in the list

Reply:
Sorry, I'm still not sure I understand here. If the regex captures, the first item in the array (0) is the whole match and the second is the result of the capture group (1). If the regex matches, then the capture group must also be defined, so I don't think we need this check as to get to this point `match.length` must equal 2.

Reply:
Fixed below